src/net/sourceforge/kolmafia/request/RelayRequest.java

```diff
@@ -3500,6 +3514,156 @@ public boolean pollForCompletion(String type) {
     return true;
   }
 
+  private void jsonError(String message) {
+    this.statusLine = "HTTP/1.1 400 Bad Request";
+    this.contentType = "application/json";
+    this.responseCode = 400;
+    this.responseText = JSON.toJSONString(Map.of("error", message));
+  }
+
+  private String underscoreName(String name) {
+    return name.replaceAll("[A-Z]", "_$0").toLowerCase();
+  }
+
+  private Value transformApiArgument(Object thing) {
+    if (thing instanceof JSONObject obj) {
+      var objectTypeObject = obj.get("objectType");
+      Type dataType;
+      if (objectTypeObject instanceof String objectType
+          && ((dataType = DataTypes.enumeratedTypes.find(objectType.toLowerCase())) != null)) {
+        var identifierStringObject = obj.get("identifierString");
+        var identifierNumberObject = obj.get("identifierNumber");
+        if (identifierStringObject instanceof String identifierString) {
+          return dataType.parseValue(identifierString, true);
+        } else if (identifierNumberObject instanceof Integer identifierNumber) {
+          return dataType.makeValue(identifierNumber, true);
+        }
+      }
+
+      return JSONValueConverter.fromJSON(obj);
```

Comment:
is this ever valid?

Reply:
I think so -- if the user does this, it's an error on their part, right? So we should explicitly error.

If you return null both here and on line 3561 that should get caught by your null check on line 3650, which I think should then error as desired.

Reply:
okay, so there is a scenario in which this is not an error: the user passes a non-trivial, non-enumerated type.

An example would be `count({"a": 1})`.

Keep this not erroring, and add that as a test, please?